### PR TITLE
CI: add option to build a scons cache docker image

### PR DESCRIPTION
**Pull request by jnewb1**
If we build a docker image that contains the scons cache, we can use --mount in buildkit to use this cache in other docker images, without it bloating the size of the image

### CODE REVIEW

**Comment by adeebshihadeh** [comment]
Might be nice to break out all those docker helper environment variables into scripts in order to make running locally easier.

File: Dockerfile.scons_cache [hunk 1]
```diff
@@ -0,0 +1,3 @@
+FROM alpine:3
+
+COPY ./scons_cache /tmp/scons_cache
```
Comment (adeebshihadeh) [inline Dockerfile.scons_cache:3]: I don't want to pollute the root with a file like this. let's either move or generate on the fly

**Comment by adeebshihadeh** [comment]
Btw, this doesn't quite seem to make sense to split up. If I understand this right, this PR on its own doesn't do anything?

**Comment by jnewb1** [comment]
> Btw, this doesn't quite seem to make sense to split up. If I understand this right, this PR on its own doesn't do anything?

yep it just adds the option to setup, but its disabled by default. I'll just close this one and move to https://github.com/commaai/openpilot/pull/29436

**Pull request closed without merging**